src/d/actor/d_a_shutter.cpp

```diff
@@ -5,80 +5,292 @@
+    if (!fopAcM_isSwitch(this, mNo)) {
+        field_0x328 = 3;
+    }
+    else {
+        field_0x328 = 2;
+        mcXyz[0].x = mcXyz[0].x - m_move_len[mType];
+        mcXyz[1].x = mcXyz[1].x + m_move_len[mType];
+    }
+    mCount = 0x1e;
+    set_mtx();
+    for (int i = 0; i < (int)ARRAY_SIZE(mMtx); i++) {
+        dComIfG_Bgsp()->Regist(mdBgW[i], this);
+        mdBgW[i]->Move();
+    }
+    if (m_open_ev_name[mType] != NULL) {
+        mOpenEventIdx = dComIfGp_evmng_getEventIdx(m_open_ev_name[mType], 0xff);
+    }
+    if (m_close_ev_name[mType] != NULL) {
+        mCloseEventIdx = dComIfGp_evmng_getEventIdx(m_close_ev_name[mType], 0xff);
+    }
+    return 1;
 }
 
 /* 000004B4-000005A0       .text _create__11daShutter_cFv */
 s32 daShutter_c::_create() {
-    /* Nonmatching */
+    fopAcM_SetupActor(this, daShutter_c);
+    mType = daShutter_prm::getType(this);
+    int result = dComIfG_resLoad(&mPhs, m_arcname[mType]);
+    if (result == cPhs_COMPLEATE_e) {
+        if (!fopAcM_entrySolidHeap(this, CheckCreateHeap, m_heapsize[mType])){
+            return cPhs_ERROR_e;
+        }
+        else {
+            Create();
+        }
+    }
+    return result;
 }
 
 /* 000005E0-000006F0       .text set_mtx__11daShutter_cFv */
 void daShutter_c::set_mtx() {
-    /* Nonmatching */
+    cXyz local_48;
+    for (int i = 0; i < (int)ARRAY_SIZE(mMtx); i++) {
+        mDoMtx_stack_c::YrotS(current.angle.y);
+        mDoMtx_stack_c::multVec(&mcXyz[i], &local_48);
+        mpModel[i]->setBaseScale(scale);
+        mDoMtx_stack_c::transS(current.pos.x + local_48.x, current.pos.y + local_48.y, current.pos.z + local_48.z);
+        mDoMtx_stack_c::ZXYrotM(current.angle.x, current.angle.y, current.angle.z);
+        mpModel[i]->setBaseTRMtx(mDoMtx_stack_c::get());
+        MTXCopy(mDoMtx_stack_c::get(), mMtx[i]);
+    }
 }
 
 /* 000006F0-00000788       .text _execute__11daShutter_cFv */
 bool daShutter_c::_execute() {
-    /* Nonmatching */
+    bool bVar1;
+    int iVar3;
+    if (mCount >= 0) {
+        mCount--;
+    }
+    demo();
+    set_mtx();
+    for (int i = 0; i < (int)ARRAY_SIZE(mMtx); i++) {
+        mdBgW[i]->Move();
+    }
+    mSwitch = fopAcM_isSwitch(this, mNo);
+    return TRUE;
 }
 
+const char* M_act_table[4] = {"WAIT", "WAIT02", "OPEN", "CLOSE"};
+
 /* 00000788-00000B14       .text shutter_move__11daShutter_cFv */
 void daShutter_c::shutter_move() {
-    /* Nonmatching */
+    int actionIndex = dComIfGp_evmng_getMyActIdx(mStaffId, (char**)M_act_table, ARRAY_SIZE(M_act_table), FALSE, 0);
+
+    float maxVel;
+    float minVel;
+
+    maxVel = m_max_speed[mType];
+    minVel = m_min_speed[mType];
+
+    float fVar4;
+    float fVar5;
+
+    switch (actionIndex) {
+        case 0: //WAIT
+        {
+            mTimer = 0xf;
+            field_0x33A = 0;
+            dComIfGp_evmng_cutEnd(mStaffId);
+            return;
+        }
+        case 1: //WAIT02
+        {
+            if (!cLib_calcTimer((u8*)&mTimer)) {
```

Comment:
```suggestion
            if (!cLib_calcTimer(&mTimer)) {
```
This indicates `mTimer`'s type is supposed to be `u8`, not `s8`.